Login and register

lan-nguyen91 opened the pull request.

do-kevin committed:
Initial commit

Init LoginModal and Registration API - sends email post request

Ignores .env

Adds axios

Styles Login modal and btn

Adds Forik

do-kevin committed:
Implements simple Formik

Styles form and input

Adds disabled styles due to Formik's isSubmitting

Adds react-router-dom

Init Landing page and 404 page

Styles NotFound page

do-kevin committed:
Sets up catch all route

Cleaner export

Removes setState, already handled by Formik

Adds query-string dependency

Adds additional forms

Styles input err messages

Init VerificationScreen

lan-nguyen91 commented on src/LoginSignup/LoginModal.js:
```diff
@@ -0,0 +1,53 @@
+					<Formik
+						initialValues={{ email: '' }}
+						onSubmit={(values) => {
+							this.setState(() => ({
```
no need binding here, formik take care of it

do-kevin replied:
Are you talking about the `this.setState()`?

lan-nguyen91 commented on src/LoginSignup/RegistrationAPI.js:
```diff
@@ -0,0 +1,20 @@
+
+export async function registerEmail(userEmail) {
+	if (typeof userEmail !== 'string') console.warn('userEmail parameter is not a string.');
+	await axios
```
let result = await fn() 

why .then if you are going to async await, if you expect error, wrap it in a try/catch

lan-nguyen91 commented on src/Misc/NotFound.js:
```diff
@@ -0,0 +1,17 @@
+			<section className="NotFound__section">
+				<p className="NotFound__section__p1">404</p>
+				<p className="NotFound__section__p2">PAGE NOT FOUND</p>
+        <p className="NotFound__section__p3">There's nothing here.</p>
```
indentation

lan-nguyen91 commented on src/Misc/NotFound.js:
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+export default function NotFound() {
```
export default = () =>   
// if you think this is cleaner

lan-nguyen91 commented on src/index.js:
```diff
@@ -0,0 +1,6 @@
+import './index.scss';
+import App from './App';
+
+ReactDOM.render(<App />, document.querySelector('#root'));
```
let add hot reload here to make your life easier

https://medium.com/@brianhan/hot-reloading-cra-without-eject-b54af352c642

lan-nguyen91 commented on src/index.scss:
```diff
@@ -0,0 +1,129 @@
+  margin: 13% auto 0;
+  background: $night1;
+
+  &__registerEmail-btn {
```
while this is all okay that you build custom CSS, and I will not stop you from doing it, there might be possibility I will throw this away, and replace with material-ui default btn

lan-nguyen91 commented on src/Misc/LandingPage.js:
```diff
@@ -0,0 +1,8 @@
+import React, { Component } from 'react';
+import LoginModal from '../LoginSignup/LoginModal';
```
let do absolute import to make your life easier
https://medium.com/@ktruong008/absolute-imports-with-create-react-app-4338fbca7e3d

do-kevin committed:
Refactor code into try-catch block

Implements HMR with react-router-dom included

Implements absolute paths after NODE_PATH=src/

lan-nguyen91 committed:
Relayout Directory and Apply Material Ui

Adds classnames dependency

Adds Material UI custom css and htmlFor

For Material UI typography deprecation warning

Replaces w/  Material UI button

do-kevin committed:
Adds verifyEmail POST req

Removes unused styles

Adds missing trailing comma
